src/repeat.js

```diff
@@ -94,17 +115,74 @@ export class Repeat {
     }
   }
 
-  processItemsByStrategy() {
+  getInnerCollection() {
+    let expression = unwrapExpression(this.sourceExpression);
+    if (!expression) {
+      return null;
+    }
+    return expression.evaluate(this.scope, null);
+  }
+
+  observeInnerCollection() {
+    let items = this.getInnerCollection();
+    if (items instanceof Array) {
+      this.collectionObserver = this.observerLocator.getArrayObserver(items);
+    } else if (items instanceof Map) {
+      this.collectionObserver = this.observerLocator.getMapObserver(items);
+    } else {
+      return false;
+    }
+    this.callContext = 'handleInnerCollectionChanges';
+    this.collectionObserver.subscribe(this.callContext, this);
+    return true;
+  }
```

Comment:
What if we added `getObserver(observerLocator, collection)` to the `CollectionStrategyLocator`?  It could use the `matchers` to find the appropriate strategy class.  Then we'd need a static method on the strategy classes that returns an observer when given the observerLocator and collection.  Is it weird to have a static method be part of the strategy interface?